# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/ipc/PCanvas.ipdl
# Commit: de6ddf5eb2a7
# Full Hash: de6ddf5eb2a7bd3d8a2f58636760e0aa16c584b9
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2023-12-05 21:29:44
# Regressor Bug: 1829026
# File Overlap Count: 1
# Description:
#   Bug 1829026 - Update CanvasTranslator to work with DrawTargetWebgl. r=aosmond
#   
#   This adds the necessary infrastructure for CanvasTranslator to allocate DrawTargetWebgl
#   instead of just allocating TextureData, and to use RemoteTextureMap to handle sending
#   the DrawTargetWebgl frames to the compositor.
# ==============================================================================

diff -r 773e7b31ac6e -r de6ddf5eb2a7 gfx/layers/ipc/PCanvas.ipdl
--- a/gfx/layers/ipc/PCanvas.ipdl	Tue Dec 05 09:37:03 2023 +0000
+++ b/gfx/layers/ipc/PCanvas.ipdl	Tue Dec 05 09:37:03 2023 +0000
@@ -11,6 +11,7 @@
 [MoveOnly] using mozilla::CrossProcessSemaphoreHandle from "mozilla/ipc/CrossProcessSemaphore.h";
 using mozilla::layers::TextureType from "mozilla/layers/LayersTypes.h";
 [MoveOnly] using mozilla::ipc::SharedMemoryBasic::Handle from "mozilla/ipc/SharedMemoryBasic.h";
+using mozilla::gfx::BackendType from "mozilla/gfx/Types.h";
 
 namespace mozilla {
 namespace layers {
@@ -29,7 +30,8 @@
    * memory handle for the ring buffer. aReaderSem and aWriterSem are handles
    * for the semaphores to handle waiting on either side.
    */
-  async InitTranslator(TextureType aTextureType, Handle aReadHandle,
+  async InitTranslator(TextureType aTextureType, BackendType aBackendType,
+                       Handle aReadHandle,
                        CrossProcessSemaphoreHandle aReaderSem,
                        CrossProcessSemaphoreHandle aWriterSem,
                        bool aUseIPDLThread);
@@ -57,6 +59,22 @@
    * Deactivate remote canvas, which will cause fall back to software.
    */
   async Deactivate();
+
+  /**
+   * Block further accelerated canvases from being created, but allow existing
+   * canvases to continue processing.
+   */
+  async BlockCanvas();
+
+  /**
+   * Notify that a remote accelerated canvas requires a fallback refresh.
+   */
+  async NotifyRequiresRefresh(int64_t aTextureId);
+
+  /**
+   * Cache the shmem of the framebuffer for snapshotting.
+   */
+  async SnapshotShmem(int64_t aTextureId, Shmem aShmem) returns (bool aSuccess);
 };
 
 } // layers